elasticsearch/index.d.ts

```diff
@@ -208,242 +369,1204 @@ declare module Elasticsearch {
+        index?: NameList;
+        type?: NameList;
+    }
+
+    export interface MTermVectorsParams extends GenericParams {
+        ids?: NameList;
+        termStatistics?: boolean;
+        fieldStatistics?: boolean;
+        fields?: NameList;
+        offsets?: boolean;
+        positions?: boolean;
+        payloads?: boolean;
+        preference?: string;
         routing?: string;
-        timeout?: Date | number;
-        timestamp?: Date | number;
+        parent?: string;
+        realtime?: boolean;
         version?: number;
-        versionType?: string;
+        versionType?: VersionType;
+        index: string;
+        type: string;
+    }
+
+    export interface PingParams extends GenericParams {
+    }
+
+    export interface PutScriptParams extends GenericParams {
+        id: string;
+        lang: string;
+        body: any;
+    }
+
+    export interface PutTemplateParams extends GenericParams {
+        id: string;
+        body: any;
+    }
+
+    export interface ReindexParams extends GenericParams {
+        refresh?: boolean;
+        timeout?: DateLike;
+        waitForActiveShards?: string;
+        waitForCompletion?: boolean;
+        requestsPerSecond?: number;
+        body: {
+            conflicts?: string;
+            source: {
+                index: string | string[];
+                type?: string | string[];
+                query?: any;
+                sort?: any;
+                size?: number;
+                remote?: {
+                    host: string;
+                    username?: string;
+                    password?: string;
+                }
+            };
+            dest: {
+                index: string;
+                version_type?: string;
+                op_type?: string;
+                routing?: string;
+                pipeline?: string;
+            };
+            script?: {
+                inline: string;
+                lang: string;
+            }
+        }
+    }
+
+    export interface ReindexResponse {
+        took: number;
+        updated: number;
+        created: number;
+        batches: number;
+        version_conflicts: number;
+        retries: {
+            bulk: number;
+            search: number;
+        }
+        throttled_millis: number;
+        failures: any[];
+    }
+
+    export interface ReindexRethrottleParams extends GenericParams {
+        requestsPerSecond: number;
+        taskId: string;
+    }
+
+    export interface RenderSearchTemplateParams extends GenericParams {
+        id: string;
     }
 
     export interface ScrollParams extends GenericParams {
-        scroll: string;
+        scroll: DateLike;
         scrollId: string;
     }
 
     export interface SearchParams extends GenericParams {
-        index?: string;
-        type?: string | string[] | number;
-        body?: any;
-        q?: string;
-        scroll?: string;
-        search_type?: string;
```

Comment:
because `search_type` is still an accepted parameter, as shown in the example https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#api-scroll